### PR TITLE
misc router fixes

### PR DESCRIPTION
**Pull request by nathanstitt**
Fixes a few issued identified in testing:
 * calendar couldn't be clicked to add assignment
 * new course wizard failed
 * task steps would cause errors when rapidly advancing

### CODE REVIEW

File: tutor/specs/screens/task/__snapshots__/milestones.spec.js.snap [hunk 1]
```diff
@@ -730,12 +730,7 @@ exports[`Reading Milestones Component matches snapshot 1`] = `
         "webview_url": "https://qa.cnx.org/contents/185cbf87-c72e-48f5-b51e-f14f21b5eabd",
         "year": 2017,
       },
-      "history": Object {
-        "createHref": [MockFunction],
-        "listen": [MockFunction],
-        "push": [MockFunction],
-        "replace": [MockFunction],
-      },
+      "history": undefined,
```
Comment (Dantemss) [inline tutor/specs/screens/task/__snapshots__/milestones.spec.js.snap]: Is this a problem?

File: tutor/src/screens/task/ux.js [hunk 2]
```diff
@@ -195,10 +193,19 @@ export default class TaskUX {
...
+      CenterControls.currentTaskStep = this.currentStep;
     }
   }
 
+  @computed get isApiPending() {
+    return get(this.currentStep, 'api.isPending', false);
+  }
+
   @computed get canGoForward() {
+    if (this.isApiPending) { return false; }
```
Comment (Dantemss) [inline tutor/src/screens/task/ux.js]: I thought we wanted them to be able to move forward while the answer is pending?

File: tutor/specs/screens/task/__snapshots__/milestones.spec.js.snap (hunk 1 above)
Comment (nathanstitt) [reply tutor/specs/screens/task/__snapshots__/milestones.spec.js.snap]: nah, thats just the snapshot

File: tutor/src/screens/task/ux.js (hunk 2 above)
Comment (nathanstitt) [reply tutor/src/screens/task/ux.js]: we did but if the step type changes then it's possible the code would start to render one type and then it'd update afterwards.  I suspect that might be part of the `map on undefined` issue

**Pull request merged**